engine/src/main/java/org/pentaho/di/job/entries/job/JobEntryJob.java

```diff
@@ -973,7 +973,8 @@ public Result execute( Result result, int nr ) throws KettleException {
               // opted to do.
               //
               if ( isPassingAllParameters() ) {
-                String parentValue = parentJob.getParameterValue( parameterNames[idx] );
+                String parentValue = Utils.isEmpty( parentJob.getParameterValue( parameterNames[idx] ) ) ? parentJob.getParameterDefault( parameterNames[idx] )
```

Comment:
I think this is wrong - as it's getting the parameter value twice (performance issue). Suggest it should say this:

String parentValue = parentJob.getParameterValue( parameterNames[idx] );
if ( Utils.isEmpty( parentValue ) ) {
  parentValue = parentJob.getParameterDefault( parameterNames[idx] );
}

- The call to parentJob.getParameterValue is only called once
- The call to parentJob.getParameterDefault is only called when the parameter value is empty